enhancements/network/on-prem-configurable-lb-dns-stack.md

```diff
@@ -0,0 +1,141 @@
+
+### CRD definition
+
+Below is a possible option for a CRD instance of this operator:
+
+
+```yaml
+apiVersion: clusterhosted.operator.io/v1alpha1
+kind: ClusterHostedConfig
+metadata:
+  name: clusterhostedconfig-sample
+spec:
+  dns: Enable
+  loadbalancer: Enable
+status:
+  apiserverinternalipowner: master-0
+  ingressipowner: worker-1
+```
+
+This manifest creates DNS and loadbalancer self-hosted stack, from the status section we can see that master-0 node currently owns api-int VIP and worker-1 node owns ingress VIP. Additional spec and status fields may be introduced based on requirements.
+
+### User Stories
+
+#### As an admin I want to disable the self-hosted on-prem LB and DNS stack.
+
+#### As an admin I want to use the self-hosted on-prem LB and DNS stack also in UPI deployments.
+
+#### As an admin I want to retrieve the self-hosted stack status through K8S API.
+
+
+
+### Upgrade / Downgrade Strategy
+
+Prior to this proposal, the self-hosted stack implementation was running through the MCO.
+The implemntation of the self-hostd stack using the new operator might lead to cases during upgrade/downgrade in which two parallel components are running the self-hosted stack. 
+
+
+It's required to verify that LB and DNS services are available in case where both MCO and the new operator run the self-hosted stack.
+
+[TODO] It's required to further understand the availability of LB and DNS services when both MCO and new operator run simultaneously. 
+
+## Alternatives
+
+Enabling/disabling of the self-hosted stack could be provided also by extending the monitor containers functionality. 
+Currently, the self-hosted stack is implemented by: Keepalived, HAProxy, MDNS-publisher and CoreDns static pods, each one of these pods includes a 
+monitor container which is responsible for rendering its configuration.
+
+It's possible to support enabling/disabling of the self-hosted stack in the following way:
+
+* Addition of new fields to the [controllerconfig MCO CR](https://github.com/openshift/machine-config-operator/blob/master/manifests/controllerconfig.crd.yaml).
+* The monitor container in each static pods should enable/disable the relevant functionality based on these fields content.
+
+The next section summarizes the pros and cons of this method.
+
+### Pros and Cons
+
+Pros:
+* No need for a separate operator.
+* All the early clustering corner cases already covered and tested.
+* Easier upgrade/downgrade path since all the components continue to run under the same operator (MCO). 
+
+Cons:
+* Self-hosted stack configuration still coupled with platform.
+* Not a typical operator controller implementation. 
+* Resources(memory/CPU) for static pods are allocated even if self-hosted stack is disabled. 
```

Comment:
This seems like kind of a dealbreaker to me. It's already possible to ignore most, if not all, of our services and deploy your own outside the cluster. The problem is that still leaves all of our stuff running and eating resources. A solution that doesn't address that isn't really a solution in my mind.

Reply:
Just to be clear, I also don't think this option is equivalent to the separate operator alternative but it has some important advantages on current implementation:
 - The services won't consume actual CPU resources. 
 - No traffic will be sent (that is important for the multicast IP traffic)